src/exchange/exchange.rs

```diff
@@ -30,19 +27,23 @@ pub struct Exchange {
     price_feeds: HashMap<String, PriceFeed>,
 }
 
-#[derive(Debug, Clone)]
-enum ExchangeError {
+#[derive(Debug, Clone, Error)]
+pub enum ExchangeError {
+    #[error("Failed to obtain asset pair")]
     FailedToObtainAssetPair,
+    #[error("Insufficient funds")]
     InsufficientFunds,
+    #[error("Failed to place order")]
     FailedToPlaceOrder,
-}
-
-impl error::Error for ExchangeError {}
+    #[error("Failed to obtain a valid price")]
+    InvalidPrice,
+    #[error("No KlineData")]
+    NoKlineDataAvailable,
```

Comment:
Just learning: what is `KlineData`?

Reply:
This is BinanceKline Data
https://github.com/DieHard073055/solid-carnival/blob/main/src/exchange/price_feed.rs#L18-L31

Reply:
Thanks for this!